syntax="proto3";
import "common.proto";

package cmq;

message SubscribeRequest
{
    string topic = 1;
}

message SubscribeResponse
{
    Status status = 1;
    string message = 2;
}

message PublishRequest
{
    string topic = 1;
    string message = 2;
}

message PublishResponse
{
    Status status = 1;
}

message PutRequest
{
    string topic = 1;
    string message = 2;
}

message PutResponse
{
    Status status = 1;
}

message GetRequest
{
    string topic = 1;
}

message GetResponse
{
    Status status = 1;
    string message = 2;
}

service broker {
    rpc Subscribe (SubscribeRequest) returns (stream SubscribeResponse);
    rpc Publish (PublishRequest) returns (PublishResponse);
    rpc Put (PutRequest) returns (PutResponse);
    rpc Get (GetRequest) returns (GetResponse);
}
